vcd/datasource_vcd_org_vdc_test.go

```diff
@@ -36,56 +35,93 @@ func TestAccVcdVdcDatasource(t *testing.T) {
-		Providers:    testAccProviders,
-		CheckDestroy: testAccCheckVdcDestroy,
-		Steps: []resource.TestStep{
-			resource.TestStep{
-				Config: configText,
-				Check: resource.ComposeTestCheckFunc(
-					testAccCheckVcdVdcExists("vcd_org_vdc."+vdcName),
-					resource.TestCheckResourceAttrPair(
-						"data."+datasourceVdc, "org", "vcd_org_vdc."+vdcName, "org"),
-					resource.TestCheckResourceAttrPair(
-						"data."+datasourceVdc, "allocation_model", "vcd_org_vdc."+vdcName, "allocation_model"),
-					resource.TestCheckResourceAttrPair(
-						"data."+datasourceVdc, "network_pool_name", "vcd_org_vdc."+vdcName, "network_pool_name"),
-					resource.TestCheckResourceAttrPair(
-						"data."+datasourceVdc, "provider_vdc_name", "vcd_org_vdc."+vdcName, "provider_vdc_name"),
-					resource.TestCheckResourceAttrPair(
-						"data."+datasourceVdc, "enabled", "vcd_org_vdc."+vdcName, "enabled"),
-					resource.TestCheckResourceAttrPair(
-						"data."+datasourceVdc, "enable_thin_provisioning", "vcd_org_vdc."+vdcName, "enable_thin_provisioning"),
-					resource.TestCheckResourceAttrPair(
-						"data."+datasourceVdc, "storage_profile.0.enabled", "vcd_org_vdc."+vdcName, "storage_profile.0.enabled"),
-					resource.TestCheckResourceAttrPair(
-						"data."+datasourceVdc, "storage_profile.0.default", "vcd_org_vdc."+vdcName, "storage_profile.0.default"),
-					resource.TestCheckResourceAttr(
-						"vcd_org_vdc."+vdcName, "metadata.vdc_metadata", "VDC Metadata"),
-					resource.TestCheckResourceAttrPair(
-						"data."+datasourceVdc, "compute_capacity.0.cpu.0.allocated", "vcd_org_vdc."+vdcName, "compute_capacity.0.cpu.0.allocated"),
-					resource.TestCheckResourceAttrPair(
-						"data."+datasourceVdc, "compute_capacity.0.cpu.0.limit", "vcd_org_vdc."+vdcName, "compute_capacity.0.cpu.0.limit"),
-					resource.TestMatchResourceAttr(
-						"data."+datasourceVdc, "compute_capacity.0.cpu.0.overhead", regexp.MustCompile(`^\d+$`)),
-					resource.TestMatchResourceAttr(
-						"data."+datasourceVdc, "compute_capacity.0.cpu.0.reserved", regexp.MustCompile(`^\d+$`)),
-					resource.TestMatchResourceAttr(
-						"data."+datasourceVdc, "compute_capacity.0.cpu.0.used", regexp.MustCompile(`^\d+$`)),
-					resource.TestCheckResourceAttrPair(
-						"data."+datasourceVdc, "compute_capacity.0.memory.0.allocated", "vcd_org_vdc."+vdcName, "compute_capacity.0.memory.0.allocated"),
-					resource.TestCheckResourceAttrPair(
-						"data."+datasourceVdc, "compute_capacity.0.memory.0.limit", "vcd_org_vdc."+vdcName, "compute_capacity.0.memory.0.limit"),
-					resource.TestMatchResourceAttr(
-						"data."+datasourceVdc, "compute_capacity.0.memory.0.overhead", regexp.MustCompile(`^\d+$`)),
-					resource.TestMatchResourceAttr(
-						"data."+datasourceVdc, "compute_capacity.0.memory.0.reserved", regexp.MustCompile(`^\d+$`)),
-					resource.TestMatchResourceAttr(
-						"data."+datasourceVdc, "compute_capacity.0.memory.0.used", regexp.MustCompile(`^\d+$`))),
+	if !usingSysAdmin() {
```

Comment:
(Following up from the comment about `fillTemplate`)
Wouldn't be more readable to have two functions instead of a giant `if`? If we split it, we can just skip one of them when the user is not sys-admin, and keep the other clean.

Reply:
updated